### PR TITLE
feat(lock): lockfiles within exclusive bucket

### PR DESCRIPTION
**Pull request by varas**
Use a different bucket for exclusively for locks.

### CODE REVIEW

File: publisher/publisher.go [hunk 1]
```diff
@@ -101,7 +101,7 @@ func main() {
 	l.Println(fmt.Sprintf("config: %v", conf))
 
 	// config validation
-	if conf.awsBucket == "" {
+	if conf.awsLockBucket == "" {
 		l.Fatal("missing 'aws_s3_bucket_name' value")
```
Comment (Noly) [inline publisher/publisher.go:105]: shouldn't be 'aws_s3_bucket_name_lock'?
Also: should we keep both? [on](https://github.com/newrelic/infrastructure-publish-action/pull/36/files#diff-20c9faf03c8712042b46f27a871aabcfc55ef92d3cab30c6f0f30669be0d2f8dR161)

```
 	viper.BindEnv("aws_s3_bucket_name")
 	viper.BindEnv("aws_s3_bucket_name_lock")
```

	
And what about named `aws_s3_lock_bucket_name` (only a suggestion, not requiried)

Comment (varas) [reply publisher/publisher.go]: there's no need to validate `aws_s3_bucket_name` env-var within publisher.go as it's not used there

Comment (Noly) [reply publisher/publisher.go]: Okey! So we only need to update the fatal message:
```
l.Fatal("missing 'aws_s3_lock_bucket_name' value")
```

Comment (varas) [reply publisher/publisher.go]: it is ok https://github.com/newrelic/infrastructure-publish-action/pull/36/files#diff-20c9faf03c8712042b46f27a871aabcfc55ef92d3cab30c6f0f30669be0d2f8dR104-R105
isn't it?

**Review by Noly — APPROVED** [review]

**Pull request merged**